src/index.ts

```diff
@@ -197,6 +199,21 @@ class DetailsDialogElement extends HTMLElement {
         toggleDetails(details, false)
       }
     })
+    this.addEventListener('submit', function(event: Event) {
+      if (!(event.target instanceof HTMLFormElement)) return
+
+      const {target} = event
+      const submitEvent = 'submitter' in event ? event as SubmitEvent : null
+      const method = submitEvent?.submitter.getAttribute('formmethod') || target.getAttribute('method')
```

Comment:
Perhaps 
```suggestion
      const method = submitEvent?.submitter?.getAttribute('formmethod') || target.getAttribute('method')
```

> If the submission was not triggered by a button of some kind, the value of submitter is null.

https://developer.mozilla.org/en-US/docs/Web/API/SubmitEvent/submitter

Reply:
Won't [?](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#description) _up_ the chain prevent null access _down_ the chain?

![Screen Shot 2021-09-15 at 11 15 14 AM](https://user-images.githubusercontent.com/2575027/133460971-ae90b692-bbd2-45b1-910e-415a25e48936.png)

Reply:
It will only work on the property or variable to which it is attached, so it does not prevent null access downward:

![Screenshot_20210915_163922](https://user-images.githubusercontent.com/118266/133464815-72da20de-e5ca-449c-961c-cb4d44fdfc12.png)

Reply:
Oh interesting! I'll push up a guard clause.